metadata-service/factories/src/main/java/com/linkedin/gms/factory/common/CacheConfig.java

```diff
@@ -74,8 +73,44 @@ public CacheManager hazelcastCacheManager() {
         .setEnabled(true)
         .setProperty("service-dns", hazelcastServiceName);
 
-    HazelcastInstance hazelcastInstance = Hazelcast.newHazelcastInstance(config);
+    return Hazelcast.newHazelcastInstance(config);
+  }
 
+  @Bean
+  @ConditionalOnProperty(name = "searchService.cacheImplementation", havingValue = "hazelcast")
+  public CacheManager hazelcastCacheManager(
+      @Qualifier("hazelcastInstance") final HazelcastInstance hazelcastInstance) {
     return new HazelcastCacheManager(hazelcastInstance);
   }
+
+  @Bean
+  @ConditionalOnProperty(name = "searchService.cacheImplementation", havingValue = "hazelcast")
+  public MapConfig defaultMapConfig() {
+    // TODO: This setting is equivalent to expireAfterAccess, refreshes timer after a get, put,
+    // containsKey etc.
+    //       is this behavior what we actually desire? Should we change it now?
+    MapConfig mapConfig = new MapConfig().setMaxIdleSeconds(cacheTtlSeconds);
+
+    EvictionConfig evictionConfig =
+        new EvictionConfig()
+            .setMaxSizePolicy(MaxSizePolicy.PER_NODE)
+            .setSize(cacheMaxSize)
+            .setEvictionPolicy(EvictionPolicy.LFU);
+    mapConfig.setEvictionConfig(evictionConfig);
+    mapConfig.setName("default");
+    return mapConfig;
+  }
+
+  @Bean
+  @ConditionalOnProperty(name = "searchService.cacheImplementation", havingValue = "hazelcast")
+  public ReplicatedMapConfig distributedThrottleMapConfig() {
```

Comment:
Nice, this probably makes sense because it has higher frequency of updates and tighter latency constraints with a low memory footprint.